src/main/scala/org/constellation/snapshotstreaming/opensearch/mapper/SnapshotMapper.scala

```diff
@@ -118,24 +122,28 @@ abstract class SnapshotMapper[F[_]: Async, S <: OriginalSnapshot] {
   private def mapTransactionRef(nodeRef: OriginalTransactionReference): TransactionReference =
     TransactionReference(nodeRef.hash.value, nodeRef.ordinal.value)
 
-  def snapshotReferredBalancesInfo(
+  def balanceDiff(
     snapshot: S,
-    info: SnapshotInfo[_]
-  ): SortedMap[Address, Balance] = {
-    val snapshotReferredAddresses = extractSnapshotReferredAddresses(snapshot)
-    val bothAddresses = snapshotReferredAddresses.source ++ snapshotReferredAddresses.destination
-    val rewardsAddresses = fetchRewards(snapshot).toList.map(_.destination)
-    val addressesToKeep = bothAddresses ++ rewardsAddresses
-    val filteredBalances = info.balances.filter { case (address, _) => addressesToKeep.contains(address) }
-
-    val srcTransactions = snapshotReferredAddresses.source
-    val setZeroBalances =
-      (srcTransactions.toSet -- info.balances.keys.toSet)
-        .map(address => address -> Balance(0L))
-        .toSortedMap
-
-    filteredBalances ++ setZeroBalances
-  }
+    prevBalances: Option[SortedMap[Address, Balance]],
+    info: SnapshotInfo[_],
+  ): SortedMap[Address, Balance] =
+    prevBalances match {
+      case Some(prev) =>
+        val changed = info.balances.filterNot { case (address, balance) =>
+          prev.get(address).exists(_ === balance)
+        }
+
+        val explicitlyZeroed = {
+          val srcTransactions = extractSnapshotReferredAddresses(snapshot).source
```

Comment:
it is not obvious to me why these accounts should be zeroed but I assume it is a detail about the currency snapshots. May be worth a comment sometime in the future

Reply:
The state (aka `SnapshotInfo` is created by applying incremental snapshots onto initial state.
One of the optimisations is that if you have an account `A` which has for example `1000 DAG` and there is a transaction `A -> B` for `1000 DAG` then the balance for `A` gets pruned instead of zeroed.
In case of node it is fine but here in block-explorer we want it to explicitly be `0`.